test/e2e/tests/test_scenarios.py

```diff
@@ -37,6 +37,63 @@ class ParameterGroupHelper(helper.ResourceHelper):
     Overrides methods as required for custom resources.
     """
 
+    def assert_expectations(self, verb: str, input_data: dict, expectations: model.ExpectDict, reference: k8s.CustomResourceReference) -> None:
+       logging.debug(f"ParameterGroup - assert_expectations()")
+       self._assert_conditions(expectations, reference, wait=False)
+       # conditions expectations met, now check current resource against expectations
+       resource = k8s.get_resource(reference)
+       self.assert_items(expectations.get("status"), resource.get("status"))
+       self.assert_items(expectations.get("spec"), resource.get("spec"))
+
+    def assert_items(self, expectations: dict, state: dict) -> None:
+        logging.debug(f"ParameterGroup - assert_items()")
+        if not expectations:
+            # nothing to assert as there are no expectations
+            return
+        if not state:
+            # there are expectations but no given state to validate
+            # following assert will fail and assert introspection will provide useful information for debugging
+            assert expectations == state
+
+        for (key, value) in expectations.items():
+            # conditions are processed separately
+            if key == "conditions":
+                continue
+            # tags are processed by custom logic
+            if key == "tags":
```

Comment:
Rather than supporting each key individually, we could do a comparison of value to state, based on the key?
Something like the following:
```python
for (key, value) in expectations.items():
  if key not in expectations:
    continue
  assert state.get(key) == value
```

I think this should work for most of the basic string-string comparisons. Possibly for tags it wouldn't work, but we could also deal with those by sorting and then comparing.

Reply:
Yes, that is what we do. For most cases, we just do `assert (key, value) == (key, state.get(key))`. For `conditions`, `tags`, and `parameterNameValues`, we do check(assert) something different.